include/lbann/utils/image.hpp

```diff
@@ -0,0 +1,75 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2016, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef LBANN_UTILS_IMAGE_HPP
+#define LBANN_UTILS_IMAGE_HPP
+
+#include "lbann/base.hpp"
+
+namespace lbann {
+
+/**
+ * Load an image from filename.
+ * @param filename The path to the image to load.
+ * @param dst Image will be loaded into this matrix, in OpenCV format.
+ * @param dims Will contain the dimensions of the image as {channels, height,
```

Comment:
To be pedantic about Doxygen use, these should really be:
```
@param[in] filename ...
@param[out] dst ...
@param[out] dims ...
```
Same comment for other `@param`s in this file.

Ideally we'd be more "functional" about the whole thing:
```
template <typename ValueT>
struct Image
{
    El::Matrix<ValueT> data_;
    std::vector<size_t> dims_;
};

Image<unit8_t> load_image(const std::string& filename);
// And/or
void load_image(const std::string& filename, Image<uint8_t>& dst);
```
But this is probably best saved for a later PR.

Reply:
That is better saved for having real tensor support. 😉

Reply:
Sure, but wouldn't it be nice if you didn't have to go change all of the code in all these different places? :wink: